### PR TITLE
Add initial websocket for stats

### PR DESCRIPTION
**Pull request by patrickelectric**

### CODE REVIEW

File: src/web/mod.rs [hunk 1]
```diff
@@ -84,6 +88,50 @@ async fn websocket_connection(socket: WebSocket, state: AppState) {
...
+    debug!("WS client connected with ID: {identifier}");
+
+    let (mut sender, mut receiver) = socket.split();
+
+    let interval_duration = tokio::time::Duration::from_secs_f32(1.0 / frequency as f32);
+    let periodic_task = tokio::spawn(async move {
+        let mut interval = tokio::time::interval(interval_duration);
+        loop {
+            interval.tick().await;
+            let json = serde_json::to_string(&hub::hub_messages_stats().await.unwrap()).unwrap();
```
Comment (joaoantoniocardoso) [inline src/web/mod.rs:122]: maybe something like:
```suggestion
            let hub_message_stats = match hub::hub_messages_stats().await {
                Ok(hub_message_stats) => hub_message_stats,
                Err(error) => {
                    warn!("Failed getting hub message stats: {error:?}");
                    continue;
                }
            }
            let json = match serde_json::to_string(&hub_message_stats) {
                Ok(json) => json,
                Err(error) => {
                    warn!("Failed to create json from Hub Message Stats: {error:?});
                    continue;
                }
            };
```

File: src/web/mod.rs [hunk 2]
```diff
@@ -84,6 +88,50 @@ async fn websocket_connection(socket: WebSocket, state: AppState) {
...
+            if sender.send(Message::Text(json)).await.is_err() {
+                break;
+            }
+        }
+    });
+    periodic_task.await.unwrap();
```
Comment (joaoantoniocardoso) [inline src/web/mod.rs:128]:
```suggestion
    if let Err(error) = periodic_task.await {
        error!("Failed finishing task Stats WebSocket task: {error:?}");
    }
```

**Review by joaoantoniocardoso — CHANGES_REQUESTED** [review]
We'll be good to go once we get rid of those unwraps.

File: src/web/mod.rs [hunk 3]
```diff
@@ -84,6 +88,50 @@ async fn websocket_connection(socket: WebSocket, state: AppState) {
...
+}
+
+async fn stats_websocket_handler(
+    ws: WebSocketUpgrade,
+    State(state): State<AppState>,
+    Query(freq_query): Query<FrequencyQuery>,
+) -> Response {
+    ws.on_upgrade(
+        |socket| async move { stats_websocket_connection(socket, state, freq_query).await },
+    )
```
[the remark this answers is not in the document]
Comment (patrickelectric) [reply src/web/mod.rs:103]: I forgot to change it after moving the query, nice!

File: src/web/mod.rs (hunk 2 above)
Comment (patrickelectric) [reply src/web/mod.rs:128]: done.

**Review by joaoantoniocardoso — APPROVED** [review]
Thanks!

**Pull request merged**